reference/docs-conceptual/AIShell/how-to/mcp-support.md

```diff
@@ -0,0 +1,159 @@
+environment.
+
+Built-in tools
+
+- `get_working_directory` - Get the current working directory of the connected PowerShell session,
+  including the provider name (e.g., `FileSystem`, `Certificate`) and the path (e.g., `C:\\`,
+  `cert:\\`).
+- `get_command_history` - Get up to 5 of the most recent commands executed in the connected
+  PowerShell session.
+- `get_terminal_content` - Get all output currently displayed in the terminal window of the
+  connected PowerShell session.
+- `get_environment_variables` - Get environment variables and their values from the connected
+  PowerShell session. Values of potentially sensitive variables are redacted.
+- `copy_text_to_clipboard` - Copy the provided text or code to the system clipboard, making it
+  available for pasting elsewhere.
+- `post_code_to_terminal` - Insert code into the prompt of the connected PowerShell session without
+  executing it. The user can review and choose to run it manually by pressing Enter.
+- `run_command_in_terminal` - This tool allows you to execute shell commands in a persistent
+  PowerShell session, preserving environment variables, working directory, and other context across
+  multiple commands.
+- `get_command_output` - Get the output of a command previously started with
+  `run_command_in_terminal`.
+
+> [!NOTE]
+> The built-in tools rely on the side-car experience with a connected PowerShell session and provide
+> enhanced context awareness and automation capabilities.
+
+Here is a simple demo showing how you can have AI Shell run commands on your behalf using the
+`run_command_in_terminal` tool:
+
+![Have the MCP run a command in terminal for you.][09]
+
+This example shows how additional context is provided to AI Shell to improve results:
+
+![Getting more context with built-in tools.][07]
+
+You can also use the `get_terminal_content` tool to get the content from the connected terminal and
+provide it to AI Shell to help it understand what you are trying to do:
+
+![Getting content from the terminal for output created before AI Shell started.][08]
+
+## Find and configure MCP servers
+
+An MCP server is an application that runs locally or is hosted remotely and accessed through an
+endpoint URL. Many MCP servers are open source and available for free. Local MCP server apps are
+typically built using Node.js and can installed using `npm`. You can find a list of available MCP
+servers at [Awesome MCP Servers][03].
+
+Before you can use an MCP server (Local or remote) you must add it to the `mcp.json` configuration
+file. In some cases, you might not need to install the local MCP before using it. However, the MCP
+server will be installed the first time you start AI Shell. This can take several minutes. For the
+best experience, preinstall the MCP server. Consult the documentation for your specific MCP server
+for installation instructions. Remote MCP servers don't need to be installed. You just need to add
+them to the `mcp.json` file.
```

Comment:
Can we link to information about the `mcp.json` configuration file? I think this is the first mention in this article.